src/platform/Tizen/WiFiManager.cpp

```diff
@@ -219,6 +344,91 @@ void WiFiManager::_ScanFinishedCb(wifi_manager_error_e wifiErr, void * userData)
     }
 }
 
+void WiFiManager::_ScanFinishedCb(wifi_manager_error_e wifiErr, void * userData)
+{
+    ChipLogProgress(DeviceLayer, "ScanFinished Callback..");
+
+    if (wifiErr == WIFI_MANAGER_ERROR_NONE)
+    {
+        ChipLogProgress(DeviceLayer, "WiFi scan finished");
+        std::vector<WiFiScanResponse> * networkScanned = new std::vector<WiFiScanResponse>();
+
+        int err = wifi_manager_foreach_found_ap(sInstance.mWiFiManagerHandle, _FoundAPOnScanCb, networkScanned);
+        if (err == WIFI_MANAGER_ERROR_NONE)
+        {
+            if (sInstance.mpScanCallback != nullptr)
+            {
+                TizenScanResponseIterator<WiFiScanResponse> iter(const_cast<std::vector<WiFiScanResponse> *>(networkScanned));
+                chip::DeviceLayer::PlatformMgr().LockChipStack();
+                sInstance.mpScanCallback->OnFinished(Status::kSuccess, CharSpan(), &iter);
+                chip::DeviceLayer::PlatformMgr().UnlockChipStack();
+                sInstance.mpScanCallback = nullptr;
+            }
+        }
+        else
+        {
+            ChipLogError(DeviceLayer, "FAIL: get scan list [%s]", get_error_message(wifiErr));
+        }
+        delete const_cast<std::vector<WiFiScanResponse> *>(networkScanned);
+    }
+    else
+    {
+        ChipLogError(DeviceLayer, "FAIL: scan WiFi [%s]", get_error_message(wifiErr));
+    }
+}
+
+bool WiFiManager::_FoundAPOnScanCb(wifi_manager_ap_h ap, void * userData)
+{
+    bool cbRet          = true;
+    int wifiErr         = WIFI_MANAGER_ERROR_NONE;
+    char * essid        = nullptr;
+    char * bssid        = nullptr;
+    int rssi            = 0;
+    int freq            = 0;
+    auto networkScanned = static_cast<std::vector<WiFiScanResponse> *>(userData);
+    std::pair<WiFiBand, int> bandInfo;
+
+    wifi_manager_security_type_e type;
+    WiFiScanResponse scannedAP;
+
+    wifiErr = wifi_manager_ap_get_essid(ap, &essid);
+    VerifyOrExit(wifiErr == WIFI_MANAGER_ERROR_NONE,
+                 ChipLogError(DeviceLayer, "FAIL: get AP essid [%s]", get_error_message(wifiErr)));
+    ChipLogProgress(DeviceLayer, "Essid Found: %s\n", essid);
+    memcpy(scannedAP.ssid, essid, sizeof(scannedAP.ssid));
+    scannedAP.ssidLen = sizeof(char);
```

Comment:
The `scannedAP.ssidLen` should be the length of the string in the `scannedAP.ssid`, not the size of `char` type. Also, in current form there is a potential out-of-bound read in case where the `essid` string returned by the `wifi_manager_ap_get_essid()` is longer than the `scannedAP.ssid` buffer size. When doing memcopy you have to check for the size of destination buffer (so there will be no overflow) and the size/length of the source buffer (not to make out-of-bound read).

Please fix that in all other places where a string obtained from a Tizen native API is copied to a fix-sized buffer.

Reply:
Changed the code to

    ssidLen = sizeof(scannedAP.ssid);
    if (ssidLen >= strlen(essid))
    {
        ssidLen = strlen(essid);
    }
    memcpy(scannedAP.ssid, essid, ssidLen);

So it will decide the copy string _ssidLen_ first to avoid any overrun or overflow of the memory.